syntax = "proto3";

import "TaskCommon.proto";

message TaskMetadata {
    string module = 1;
    string function = 2;
    int32 args = 3;
    repeated string kwargs = 4;
    string task_id = 5;
    string _class = 6;
}


message TaskReceived {
    Status status = 1;
    string task_id = 2;
}

enum ArgumentType {
    ARGUMENT = 0;
    KEYWORD_ARGUMENT = 1;
}

message ArgumentMetadata {
    ArgumentType arg_type = 1;
    string task_id = 2;
    oneof arg_sequence {
        string name = 3;
        int32 num = 4;
    }
}

message TaskArgument {
    oneof task_arg {
        ArgumentMetadata arg_seq = 1;
        bytes arg = 2;
    }
}

message ExecTaskRequest {
    string task_id = 1;
}

message TaskResponse {
    oneof task_response {
        Status status = 1;
        bytes result = 2;
    }
}

